scripts/stubsabot.py

```diff
@@ -470,11 +499,91 @@ async def analyze_diff(
     return DiffAnalysis(py_files=py_files, py_files_stubbed_in_typeshed=py_files_stubbed_in_typeshed)
 
 
-async def determine_action(distribution: str, session: aiohttp.ClientSession) -> Update | NoUpdate | Obsolete:
+def _add_months(date: datetime.date, months: int) -> datetime.date:
+    month = date.month - 1 + months
+    year = date.year + month // 12
+    month = month % 12 + 1
+    day = min(date.day, calendar.monthrange(year, month)[1])
+    return datetime.date(year, month, day)
+
+
+def obsolete_more_than_6_months(distribution: str) -> bool:
+    try:
+        with metadata_path(distribution).open("rb") as file:
+            data = tomlkit.load(file)
+    except FileNotFoundError:
+        raise NoSuchStubError(f"Typeshed has no stubs for {distribution!r}!") from None
+
+    obsolete_since = data["obsolete_since"]
+    if not obsolete_since:
+        return False
+
+    assert type(obsolete_since) is String
+    comment: str | None = obsolete_since.trivia.comment
+    if not comment:
+        return False
+
+    release_date_string = comment.removeprefix("# Released on ")
+    release_date = datetime.date.fromisoformat(release_date_string)
+    remove_date = _add_months(release_date, POLICY_MONTHS_DELTA)
+    today = datetime.datetime.now(tz=datetime.timezone.utc).date()
```

Comment:
```suggestion
    today = datetime.date.today()
```

(Should be UTC anyway in CI, but hasn't to be perfect.)

Reply:
https://docs.astral.sh/ruff/rules/call-date-today/

Reply:
The description of that ruff rule is wrong, making the whole rule nonsensical. `datedate.date.today()` doesn't return a `datetime` object, but a `date` object. `date` objects have no timezone attribute, and are neither naïve nor timezone-aware.